src/gui/mzroll/isotopeswidget.cpp

```diff
@@ -96,6 +96,10 @@ void IsotopeWidget::setPeakGroupAndMore(PeakGroup *grp, bool bookmarkflg)
 		isotopeParameters->_group = grp->parent;
 	}
 
+	//TODO: move bookmarking functionality out of isotopeWidget
+	if (bookmarkflg)
+		pullIsotopes(isotopeParameters->_group);
+
 	//select first sample if no peak or sample is selected
 	if (!_selectedSample)
```

Comment:
@sahil21 Shift-drag just needs pullIsotopes thread to be called. This was an issue earlier because the function used to return before starting that thread. if there is no peak in the selected sample, it will still populate all isotopes for the group, but the isotopeWidget for that sample will be blank, which it should be. 

I'll move it out of isotopeWidget in 0.4.1 or 0.5.0.